src/gpg2john.c

```diff
@@ -820,8 +821,14 @@ string_to_key(void)
 		// hash_algs(Getc());
 		m_hashAlgorithm = Getc();
 		if (gpg_dbg) fprintf(stderr, "\t\tSalt - ");
-		if (gpg_dbg)
-			dump(8);
+		if (gpg_dbg) {
+			//FIXED: //if using dump(8) in debug mode, m_salt is not set and salt is not added to outputted hash
+			//dump(8);
+			give(8, m_salt, sizeof(m_salt));
+			for (i = 0; i < 8; ++i) {
+				fprintf(stderr, "%02x", m_salt[i]);
+			}
+		}
 		else
 			give(8, m_salt, sizeof(m_salt));
```

Comment:
I think both instances of the `give` call should ideally be replaced with one just before the `if`. And no need to clutter the code with comments about past bugs, the differences between these functions are obvious enough not to require that.